incubator/distribution/templates/distribution-deployment.yaml

```diff
@@ -0,0 +1,116 @@
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+spec:
+  replicas: {{ .Values.distribution.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "distribution.name" . }}
+      release: {{ .Release.Name }}
+      component: {{ .Values.distribution.name }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "distribution.name" . }}
+        component: {{ .Values.distribution.name }}
+        release: {{ .Release.Name }}
+    spec:
+    {{- if .Values.imagePullSecrets }}
+      imagePullSecrets:
+      - name: {{ .Values.imagePullSecrets }}
+    {{- end }}
+      initContainers:
+      - name: "init-data"
+        image: "{{ .Values.initContainerImage }}"
+        command:
+        - '/bin/sh'
+        - '-c'
+        - >
+          until nc -z -w 2 {{ .Release.Name }}-mongodb 27017 && echo {{ .Release.Name }}-mongodb ok; do sleep 2; done;
+          until nc -z -w 2 {{ .Release.Name }}-redis {{ .Values.redis.master.port }} && echo {{ .Release.Name }}-redis ok; do sleep 2; done;
+      containers:
+      - name: {{ .Values.distribution.name }}
+        image: '{{ .Values.distribution.image.repository }}:{{ .Values.distribution.image.version }}'
+        imagePullPolicy: {{ .Values.distribution.image.imagePullPolicy }}
+        ports:
+        - containerPort: {{ .Values.distribution.internalPort }}
+          protocol: TCP
+        env:
+        - name: DEFAULT_JAVA_OPTS
+          value: '-Ddistribution.home={{ .Values.distribution.persistence.mountPath }} -Dcom.sun.management.jmxremote.authenticate=false
+                  -Dcom.sun.management.jmxremote.ssl=false -Duser.timezone=UTC
+                  {{- if .Values.distribution.javaOpts.xms }}
+                  -Xms{{ .Values.distribution.javaOpts.xms }}
+                  {{- end}}
+                  {{- if .Values.distribution.javaOpts.xmx }}
+                  -Xmx{{ .Values.distribution.javaOpts.xmx }}
+                  {{- end}}
+                  -Dspring.profiles.active=production'
+        - name: mongo_connectionString
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "distribution.fullname" . }}-mongo-connection
+              key: mongo_connectionString
+        - name: audit_mongo_connectionString
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "distribution.fullname" . }}-mongo-connection
+              key: audit_mongo_connectionString
+        - name: redis_connectionString
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "distribution.fullname" . }}-redis-connection
+              key: redis_connectionString
+        - name: BT_ARTIFACTORY_URL
+          value: {{ .Values.distribution.env.artifactoryUrl | quote }}
+        - name: BT_SERVER_URL
+          value: {{ .Values.distribution.env.btServerUrl | quote }}
+        {{- if .Values.distribution.env.artifactoryEdge1Url }}
+        - name: artifactory_edge_1_url
```

Comment:
Just wondering. It's kind of uncommon to have env vars lowercase. If case doesn't matter, I'd suggest you make them all uppercase.

Reply:
It's Case sensitive. Will get fixed with the future release of our product.